bin/i18n/i18n_script_utils.rb

```diff
@@ -9,6 +10,35 @@
 CODEORG_MARKDOWN_CONFIG_FILE = File.join(File.dirname(__FILE__), "codeorg_markdown_crowdin.yml")
 CODEORG_MARKDOWN_IDENTITY_FILE = File.join(File.dirname(__FILE__), "codeorg_markdown_credentials.yml")
 
+# Output the given data to YAML that will be consumed by Crowdin. Includes a
+# couple changes to the default `data.to_yaml` serialization:
+#
+#   1. Don't wrap lines. This is an optional feature provided by yaml, intended
+#      to make human editing of the serialized data easier. Because this data
+#      is only managed programmatically, we avoid wrapping to make the git
+#      diffs smaller and change detection easier.
+#
+#   2. Quote 'y' and 'n'. Psych intentionally departs from the YAML spec for
+#      these strings: https://github.com/ruby/psych/blob/8e880f7837db9ed66032a1dddc85444a1514a1e3/test/psych/test_boolean.rb#L21-L35
+#      But Crowdin sticks strictly to the YAML spec, so here we add special
+#      logic to ensure that we conform to the spec when outputting for Crowdin
+#      consumption.
+#      See https://github.com/gvvaughan/lyaml/issues/8#issuecomment-123132430
+def to_crowdin_yaml(data)
+  ast = Psych.parse_stream(Psych.dump(data))
+
+  # Make sure we treat the strings 'y' and 'n' as strings, and not bools
+  yaml_bool = /^(?:y|Y|n|N)$/
+  ast.grep(Psych::Nodes::Scalar).each do |node|
+    if yaml_bool.match node.value
+      node.plain = false
+      node.quoted = true
+    end
+  end
+
+  return ast.yaml(nil, {line_width: -1})
```

Comment:
@Hamms What does this line do?

Reply:
converts the ast to a string containing the yaml representation of the content, without line-wrapping long strings.

https://github.com/ruby/psych/blob/a6f8972e1d9e277997a49908bc555f1fdfded078/lib/psych/nodes/node.rb#L54-L64